apps/src/music/views/PanelContainer.jsx

```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {useSelector} from 'react-redux';
+import classNames from 'classnames';
+import moduleStyles from './panelContainer.module.scss';
+
+/**
+ * A container for a top-level panel with a header.  The content of the panel
+ * is provided as children.  Whether the header is shown is determined by an
+ * external state value.  The panel container always occupies the full size of
+ * its parent; this means that the main scene is responsible for allocating
+ * the layout of all the panels.
+ */
+const PanelContainer = ({id, headerText, children}) => {
+  const isHeadersShowing = useSelector(state => state.music.isHeadersShowing);
+
+  return (
+    <div
+      className={classNames('panelContainer', moduleStyles.panelContainer)}
+      id={id}
+    >
+      {isHeadersShowing && (
+        <div
+          className={classNames(
+            'panelContainerHeader',
```

Comment:
Why do we need these class names along with the module class names?

Reply:
Oh, I wanted to have readable identifiers for more of the DOM elements to make it easy to work in the browser's Elements tab, so many of them have new `id` entries.  However, for elements that appear more than once on a page, we shouldn't have duplicate IDs, so I used `class` instead.

Reply:
Ah okay interesting!